base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -896,25 +899,70 @@ private X509CertImpl issueX509Cert(
                     if ( ctPoison == null) {
+                        // this just means that CT is enabled, but profile doen'st,
+                        // which we honor
                         logger.debug(method + " ctPoison not found, no CT processing needed");
                     } else {
                         logger.debug(method + " ctPoison found");
+                        // debug print with poison ext
+                        // printExtensions(exts);
+
+                        // remove then put back to assure relocating poison ext
+                        // to end of the map
+                        exts.delete("1.3.6.1.4.1.11129.2.4.3");
+                        // debug print without poison ext
+                        // printExtensions(exts);
+                        certi.delete(X509CertInfo.EXTENSIONS);
+                        certi.set(X509CertInfo.EXTENSIONS, exts);
+
+                        // tbsCert_save_s is the tbsCert without poison ext;
+                        // Save for debugging later
+                        byte[] tbsCert_save = certi.getEncodedInfo(true);
+                        String tbsCert_save_s = 
+                            CryptoUtil.base64Encode(tbsCert_save);
+                        //logger.debug(method + "tbsCert without poison="
+                        //        + tbsCert_save_s);
+
+                        // now add back
+                        exts.set("1.3.6.1.4.1.11129.2.4.3", ctPoison);
+                        //printExtensions(exts);
+                        certi.delete(X509CertInfo.EXTENSIONS);
+                        certi.set(X509CertInfo.EXTENSIONS, exts);
+
                         logger.debug(method + " About to ca.sign CT pre-cert.");
                         cert = ca.sign(certi, algname);
 
                         /*
                          * remove the poison extension from certi
                          */
                         exts.delete("1.3.6.1.4.1.11129.2.4.3");
+                        logger.debug(method + " ctPoison deleted");
+                        // debug print without poison ext
+                        // printExtensions(exts);
                         certi.delete(X509CertInfo.EXTENSIONS);
                         certi.set(X509CertInfo.EXTENSIONS, exts);
                         /*
-                         * This tbsCert does not contain poison ext.
-                         * It is intended to be used for verifying
-                         * the SCT signature in the CT response,
+                         * tbsCert is the tbsCert after
+                         *   - poision ext was deleted,
+                         *   - then added back,
+                         *   - then deleted again
+                         * It can be used for comparison with
+                         * tbsCert_save_s to check if things get
+                         * out of order for debugging purpose.
```

Comment:
By the term "order", if you are referring to the order of extensions within the `exts` object the current implementation of [`CertificateExtension`](https://github.com/dogtagpki/jss/blob/master/org/mozilla/jss/netscape/security/x509/CertificateExtensions.java#L111) uses `Hashtable`, which does not guarantee the order.

As you mentioned earlier, using `LinkedHashMap` in JSS would preserve the order of the extensions.

Reply:
Yes.  As I explained in my comment, I do not wan to pose dependency on JSS yet as I'm just doing this for @frasertweedale to help taking a look quickly without waiting for JSS build.  As you can see, the code that supports LinkedHashMap has been commented out temporarily for now.

Reply:
Once you push the changes to JSS, the CI does the JSS builds automatically (in COPR). This new build will be automatically pulled into this CI execution (ie) you don't have to wait for official jss builds :-)

Reply:
oh? I did not know that!!  In that case, I'd rather switch over to LinkedHashMap on both sides (JSS and PKI).  The build magical world never seizes to amaze me. ;-)